File: random/validator.c
#include "random.h"
#include <stdlib.h>
#include <stdio.h>
#include <math.h>


int main(int argc, char * argv[]) {
	int i = 0;
	float f;
	int32_t out;
	int32_t seed;
	int32_t spec;
	int32_t test;
	if(argc != 3) {
		printf("Usage: %s <SPEC number> <test number>", argv[0]);
		exit(1);
	}
	spec = atoi(argv[1]);
	test = atoi(argv[2]);
	
	seed = calculate_seed(spec,test);
	math_randomseed(seed);
	for(i=0; i < 39; i++) { printf("%d\n", math_random(1,3)); }
	for(i=0; i < 39; i++) { printf("%d\n", math_random(1,999)); }
	for(i=0; i < 39; i++) { printf("%d\n", math_random(-3,-1)); }
	for(i=0; i < 39; i++) { printf("%d\n", math_random(-99,999)); }
	return 0;
}

File: random/validator.lua
#! /usr/bin/lua

require "random"

if #arg ~= 2 then
	print("Usage:", arg[0], "<SPEC number> <test number>")
	os.exit(1)
end

spec = tonumber(arg[1])

test = tonumber(arg[2])
if test < 1 or test > 3 then
	print("Test number is unlikely to be outside of the range 1 through 3.", arg[2], "was received.")
	os.exit(1)
end
 
seed = calculate_seed(arg[1], test)
math_randomseed(seed)
for i=1,39 do print(math_random(1,3)) end
for i=1,39 do print(math_random(1,999)) end
for i=1,39 do print(math_random(-3,-1)) end
for i=1,39 do print(math_random(-99,999)) end
